src/main/java/nu/studer/teamcity/buildscan/internal/CustomDatastoreBuildScanDataStore.java

```diff
@@ -0,0 +1,83 @@
+package nu.studer.teamcity.buildscan.internal;
+
+import com.google.common.collect.Maps;
+import jetbrains.buildServer.Build;
+import jetbrains.buildServer.serverSide.*;
+import jetbrains.buildServer.util.StringUtil;
+import nu.studer.teamcity.buildscan.BuildScanReference;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+
+public final class CustomDatastoreBuildScanDataStore extends BuildServerAdapter implements ReadOnlyBuildScanDataStore {
```

Comment:
I would prefer if we extracted all clean-up logic into a separate class.

Reply:
I did not integrate the deletion logic into master.

Reply:
I will move the whole cleanup-logic into a separate class.